docs/rfcs/0001-designing-in-the-open.md

```diff
@@ -0,0 +1,139 @@
+---
+title: "Designing in the Open"
+weight: 1
+---
+
+# Designing in the Open
+
+* Date: 2021-11-02
+* Author: Robert Fratto (@rfratto)
+* PR: [grafana/agent#1055](https://github.com/grafana/agent/pull/1055)
+
+## Summary
+
+Traditionally, design proposals to Grafana Agent have been done in private by
+default, and then only made public after internal consensus was reached.
+This causes a few problems:
+
+* Community concerns are less likely to have an impact as core maintainers have
+  already agreed on a solution.
+* Initial design is gated to Grafana Labs employees.
+* Historical design docs with context and discussions become hard to find both
+  internally and even harder publicly.
+
+In the spirit of building a broader community, this document proposes inverting
+the design process to be public by default.
+
+## Goals
+
+* Outline a process to support public-first design documents
+* Lower the barrier to entry to becoming a maintainer that is not employed at
+  Grafana Labs
+* Encourage a community to contribute input to design documents
+* Encourage a community to contribute their own design documents
+
+## Non-Goals
+
+* Make everything public. If something can't be public for confidentiality
+  reasons, it can remain internal to the company.
+* Identify strict rules for when an RFC or issue is appropriate.
+
+## Terminology
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL
+NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED",  "MAY", and
+"OPTIONAL" in this document are to be interpreted as described in
+[RFC 2119](https://datatracker.ietf.org/doc/html/rfc2119).
+
+## Proposal
+
+Public propsals come in two forms:
+
+* RFCs (i.e., this document) for larger changes
```

Comment:
Should RFC be renamed to something else to uniquely identify what project an RFC refers to? RFC seems to be a bloated term now and I think most people read RFC and think IETF.

(By most people I mean "me," I think IETF when I read RFC).

Reply:
I also tend to associate RFC with the IETF as well. Joyent changed the terminology of their process for exactly this reason, which makes sense to me: 'We use the term "Requests for Discussion" in lieu of "Requests for Comments" to avoid conflation with the IETF construct -- and the more formal writing that it has come to represent.' 

Did you have another name in mind?

Reply:
Nope, but it's also worth noting that if this process leaks to other Grafana-hosted projects, we might have the same problem multiple times. 

There's always something like RFC-Agent-0001 🙈